Fixes and improvements upon issues #500, #79, #465 and #41

viniciusarre opened the pull request.

viniciusarre commented on client/src/helpers/test/ClassList.spec.js:
```diff
@@ -0,0 +1,52 @@
+        'Download all assignment evaluation notes'
+      )
+    })
+    //   test('does clicking the button show the modal', () => {
```
@bryan-gilbert 
This is the part I asked you about unit tests in general, I am not sure what should be tested when it comes to the export as csv button functionality. I could definitely use some feedback on it! Thank you!

bryan-gilbert replied:
this is good enough for now.

bryan-gilbert commented on client/src/helpers/test/ClassList.spec.js:
```diff
@@ -0,0 +1,52 @@
+import { library } from '@fortawesome/fontawesome-svg-core'
+import {
+  faDownload,
+} from '@fortawesome/free-solid-svg-icons'
```
Do we need the font here?  Please remove if it is not needed.

viniciusarre replied:
We do. Since the proper component is used. If we don't then the test will throw an error regarding the unidentified component

bryan-gilbert commented on client/src/app/components/AppDialogShell.vue:
```diff
@@ -180,7 +181,7 @@ export default {
 .dialog-body {
   overflow-y: auto;
-  max-height: calc( 90vh - 20vh );
+  min-height: 40vh;
```
It seems the min-height is added to allow room for the date picker. Yes?  Yet why do we need to remove the max-height rule which limits the dialog to the size of the display and forces scroll bars inside the dialog for content that reaches below the fold.

bryan-gilbert commented on client/src/helpers/test/__snapshots__/ClassList.spec.js.snap:
```diff
@@ -0,0 +1,51 @@
+
+exports[`ClassName test Basic testing renders as expected 1`] = `
+<div>
+  <div class="classlist-header-group">
```
Snapshot tests are good to have yet we need to have the means to easily update the snapshots.  Soon, we need to add a script to the package.json with a command that can regenerate the snapshots after intentionaly changes.

bryan-gilbert replied:
no need to make any changes for now regarding snapshots

viniciusarre replied:
Noted. Thank you for the feedback!

bryan-gilbert commented on client/src/inside/layout/LayoutEhr.vue:
```diff
@@ -13,10 +13,10 @@
+    input(class="checkbox", type="checkbox", v-model="showingSpecial")
+    div(v-show="showingSpecial")
+      ehr-special
+    app-footer
```
Need to keep that input below the footer.
Don't make any changes for now.  We can un-fix this in another PR.

viniciusarre commented:
In order to review the issues which I have worked on, please, check the following files: 

- For issue #79, I have edited both design files, which implement the footer `client/src/inside/layout/LayoutEhr.vue` and `client/src/outside/layout/LayoutOutside.vue` by implementing the suggested flexbox solution;

- As for  #500, I have implemented it in `client/src/app/components/AppHeader.vue`, by changing the proper position of the header, by making it sticky and improving on its `z-index`;

- As for #465, I have added a `min-height` css constraint, since the problem was happening due to an automatic sizing of the `.dialog-body` class. With a fixed `min-height`, it is easy to predict that it would be visible in the entire screen, without the need of scrolling. It was implemented in `client/src/app/components/AppDialogShell.vue`;

- As for #41, I have adapted the CSV exporting logic, which had previously been implemented in another component. It has been implemented in `client/src/outside/components/ClassList.vue`, as well as some work towards a unit test for the same component, available here: `client/src/helpers/test/ClassList.spec.js`.
